lib/src/main/java/com/auth0/msg/AbstractMessage.java

```diff
@@ -0,0 +1,180 @@
+package com.auth0.msg;
+
+import com.auth0.jwt.algorithms.Algorithm;
+import org.apache.commons.codec.binary.Hex;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class AbstractMessage implements Message {
+    private Map<String, Object> claims;
+    private String input;
+    private Error error = null;
+    private boolean verified = false;
+
+    protected AbstractMessage(Map<String, Object> claims) {
+        this.claims = claims;
+    }
+
+    /**
+     * Returns a hashmap representation of the contents of the urlEncoded string
+     * which is passed in as a parameter
+     *
+     * @param urlEncoded the urlEncoded String representation of a message
+     * @return a map of the key value pairs encoded in the string parameter
+     */
+    private static Map<String, Object> claimsFromUrlEncoded(String urlEncoded) throws Exception {
```

Comment:
Having a private static method in an abstract class doesn't make sense. Subclasses will not see this method at all. If it is an utility method - create a new class MessageUtil and there you add it as a public static method.

Reply:
done